flink-libraries/flink-table/src/main/java/org/apache/flink/api/table/sql/calcite/FlinkRelNode.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.api.table.sql.calcite;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.DataSet;
+
+public interface FlinkRelNode<OUT> extends RelNode {
+	
+	TypeInformation<OUT> getType();
+	
+	String getName();
+	
+	DataSet<OUT> translateToPlan();
```

Comment:
`translateToPlan()` should be the only `public` method of all `FlinkRelNodes`, IMO. The other methods can be `private`.
`translateToPlan()` will require the information of the other private methods in order to create the corresponding DataSet operators but it is not necessary to expose that information.